### PR TITLE
Add hive style external_location to iceberg tables

### PR DESCRIPTION
**Pull request by alexey-fin**
Related to #2501

### CODE REVIEW

**Comment by lxynov** [comment]
Two preliminary comments:
1. Could you sign the CLA as the cla-bot suggested?
2. There is a checkstyle failure in `presto-iceberg`. Could you fix that?

**Comment by alexey-fin** [comment]
@lxynov signed cla & fixed import order

**Comment by kfirt** [comment]
Can you please merge this one?
(Or do we need to fix anything else?)

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java [hunk 1]
```diff
@@ -26,13 +26,15 @@
...
 import static io.prestosql.spi.session.PropertyMetadata.enumProperty;
+import static io.prestosql.spi.session.PropertyMetadata.stringProperty;
 import static io.prestosql.spi.type.VarcharType.VARCHAR;
 import static java.util.Locale.ENGLISH;
 
 public class IcebergTableProperties
 {
     public static final String FILE_FORMAT_PROPERTY = "format";
     public static final String PARTITIONING_PROPERTY = "partitioning";
+    public static final String EXTERNAL_LOCATION_PROPERTY = "external_location";
```
Comment (electrum) [inline presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java]: Let's name this `location` to match the upcoming change to the Hive connector in #1282. This also makes more sense, as we are simply specifying the location and not creating an external table (which doesn't exist as a concept in Iceberg).

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java [hunk 2]
```diff
@@ -57,6 +59,11 @@ public IcebergTableProperties(IcebergConfig icebergConfig)
                                 .map(name -> ((String) name).toLowerCase(ENGLISH))
                                 .collect(toImmutableList()),
                         value -> value))
+                .add(stringProperty(
+                        EXTERNAL_LOCATION_PROPERTY,
+                        "File system location URI for external table",
```
Comment (electrum) [inline presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java]: Remove "external" here

File: presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java [hunk 3]
```diff
@@ -76,4 +83,9 @@ public static FileFormat getFileFormat(Map<String, Object> tableProperties)
         List<String> partitioning = (List<String>) tableProperties.get(PARTITIONING_PROPERTY);
         return partitioning == null ? ImmutableList.of() : ImmutableList.copyOf(partitioning);
     }
+
+    public static String getExternalLocation(Map<String, Object> tableProperties)
```
Comment (electrum) [inline presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableProperties.java]: Rename the method to `getLocation`

**Comment by electrum** [comment]
When updating the PR, please squash into a single commit titled

> Allow specifying table location for Iceberg tables

**Pull request closed without merging**